src/ducks/SANCharts/Charts.js

```diff
@@ -456,6 +457,8 @@ class Charts extends React.Component {
         metrics.some(({ key }) => key === value) || !isAnomaly
     )
 
+    events = setColorByDayRating(events)
```

Comment:
No point for making `n` more iterations and mutating the object, when a correct color can be assigned inside loop on the `LOC 612`